android/src/main/kotlin/com/mparticle/mparticle_flutter_sdk/MparticleFlutterSdkPlugin.kt

```diff
@@ -135,6 +138,69 @@ class MparticleFlutterSdkPlugin: FlutterPlugin, MethodCallHandler {
           jsonObj.put(entry.key.toString(), value)
         }.let { result.success((it ?: JSONObject()).toString()) }
       "logCommerceEvent" -> this.logCommerceEvent(call, result)
+      "getGDPRConsentState" -> getUser(call, result)?.let { user ->
+        user.consentState.gdprConsentState
+          .toList()
+          .fold(JSONObject()) { json, (key, value) ->
+            json.put(key, value.toJSON())
+          }
+          .let {
+            result.success(it.toString())
+          }
+      }?: result.error(TAG, "User not found", null)
+      "addGDPRConsentState" -> getUser(call, result)?.let { user ->
+        val consentState = call.toGDPRConsent(result)
+        if (consentState == null) {
+          result.error(TAG, "ConsentState should not be null", null)
+          return
+        }
+        ConsentState.withConsentState(user.consentState)
+          .addGDPRConsentState(consentState.first, consentState.second)
+          .build()
+          .let {
+            user.setConsentState(it)
+          }
+        result.success(true)
+      }
+      "removeGDPRConsentState" -> {
+        getUser(call, result)?.let { user ->
+          val purpose = call.argument<String>("purpose")
+          if (purpose != null) {
+            ConsentState.withConsentState(user.consentState)
+              .removeGDPRConsentState(purpose)
+              .build()
+              .let { user.setConsentState(it) }
+            result.success(true)
```

Comment:
ditto comment to `result.success(true)` being in the `.let` above

Reply:
same comment as above but, ugh, I didn't realize the `let` here is a one-liner and the one above isn't...I'm gonna fix that 😆